Sources/Web3Core/EthereumNetwork/Request/APIRequest+UtilityTypes.swift

```diff
@@ -24,5 +24,30 @@ struct RequestBody: Encodable {
     var id = Counter.increment()
 
     var method: String
-    var params: [RequestParameter]
+    var params: [Encodable]
+
+    enum CodingKeys: String, CodingKey {
+        case jsonrpc
+        case id
+        case method
+        case params
+    }
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(jsonrpc, forKey: .jsonrpc)
+        try container.encode(id, forKey: .id)
+        try container.encode(method, forKey: .method)
+
+        var paramsContainer = container.superEncoder(forKey: .params).unkeyedContainer()
+        try params.forEach { a in
+            try paramsContainer.encode(a)
+        }
+    }
+
+    public var encodedBody: Data {
+         // this is safe to force try this here
+         // Because request must failed to compile if it not conformable with `Encodable` protocol
```

Comment:
```suggestion
         // Safe to use force-try because request will fail to
         // compile if it's not conforming to the `Encodable` protocol.
```